packages/strapi/lib/middlewares/public/index.js

```diff
@@ -114,5 +133,40 @@ module.exports = strapi => {
         })
       );
     },
+
+    template: async (data, isInitialised) => {
+      // Allowed expressions to avoid data leaking.
+      const allowedExpression = [
+        'strapi.config.info.version',
+        'strapi.config.info.name',
+        'strapi.config.admin.url',
+        'strapi.config.environment',
+        'serverTime',
+        'isInitialised',
+      ];
+
+      // Populate values to object.
+      const objectWithValues = allowedExpression.reduce((acc, key) => {
+        switch (key) {
+          case 'serverTime':
+            acc[key] = new Date().toUTCString();
+
+            break;
+          case 'isInitialised':
+            acc[key] = isInitialised;
+
+            break;
+          default: {
+            acc[key] = _.get(strapi, key.replace('strapi.', ''), '');
+          }
+        }
+
+        return acc;
+      }, {});
+
+      const templatedIndex = _.template(data);
```

Comment:
you should compile it only once at initialization to avoid recompiling on every request